packages/devtools_app/lib/src/app.dart

```diff
@@ -265,7 +267,11 @@ class DevToolsAppState extends State<DevToolsApp> with AutoDisposeMixin {
         ],
         builder: (_, __, child) {
           final screens = _visibleScreens()
-              .where((p) => embed && page != null ? p.screenId == page : true)
+              .where(
+                (p) => embedMode == EmbedMode.embedOne && page != null
```

Comment:
consider pulling this into a function (although I can't think of a good function name - `where(onlyEmbeddedScreenIfSingleEmbed)` ?)